src/frequenz/sdk/model/_manager.py

```diff
@@ -0,0 +1,98 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Load, update, monitor and retrieve machine learning models."""
+
+import asyncio
+import logging
+import pickle
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Generic, TypeVar, cast
+
+from frequenz.channels.file_watcher import EventType, FileWatcher
+
+from frequenz.sdk._internal._asyncio import cancel_and_await
+
+_logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+
+
+@dataclass
+class _Model(Generic[T]):
+    """Represent a machine learning model."""
+
+    data: T
```

Comment:
I wonder whether we want to be more specific about the model implementation. For instance the `data` will most likely be more complex, e.g. a dictionary that holds a model, feature names and other metadata.

In the current version, there is nothing in the `_Model` neither in the `ModelManager` that is "model"-specific. In fact this module is a `PickleManager`. Don't think it's a bad thing, but maybe we could be more clear about this if we keep the generic approach.

Reply:
I agree but think we don't need to do it right now. Not calling it `PickleManager` allows us to grow in time with model-specific stuff.